tests/cases/fourslash/codeFixAddMissingAwaitInReturn12.ts

```diff
@@ -0,0 +1,41 @@
+/// <reference path="fourslash.ts" />
+
+// @strict: true
+// @target: esnext
+// @lib: esnext
+
+//// async function inner() {
+////   if (Math.random() > 0.5) {
+////     throw new Error("Ooops");
+////   }
+////   return 42;
+//// }
+////
+//// async function outer() {
+////   try {
+////     [|return|] Math.random() > 0.5 ? inner() : 5;
+////   } catch (e) {}
+//// }
+
+verify.getSuggestionDiagnostics([{
+    message: ts.Diagnostics.This_may_need_await_keyword_Otherwise_the_enclosing_try_statement_won_t_handle_this.message,
+    code: 80011,
+    range: test.ranges()[0],
+}]);
+verify.codeFix({
+    description: ts.Diagnostics.Add_missing_await.message,
+    index: 0,
+    newFileContent:
+`async function inner() {
+  if (Math.random() > 0.5) {
+    throw new Error("Ooops");
+  }
+  return 42;
+}
+
+async function outer() {
+  try {
+    return await (Math.random() > 0.5 ? inner() : 5);
```

Comment:
This should actually produce this:
```suggestion
    return Math.random() > 0.5 ? await inner() : 5;
```

The current implementation is a draft and it's not yet fully implemented for more complex cases. The problem with it is that for the code like this:
```ts
return Math.random() > 0.5 ? await inner1() : inner2();
```

the compiler shouldn't produce smth like this
```ts
return await (Math.random() > 0.5 ? await inner1() : inner2());
```

but rather:
```ts
return Math.random() > 0.5 ? await inner1() : await inner2();
```